### ISSUE TITLE
Hint when using a static method on an instance is wrong

### ISSUE THREAD

**Issue by jdm** [rust-lang/rust#29121]
I wrote `Box::new(telemetry_t::new()).into_raw()` and got this output:
```
src/lib.rs:41:34: 41:44 error: no method named `into_raw` found for type `Box<telemetry_t>` in the current scope
src/lib.rs:41     Box::new(telemetry_t::new()).into_raw()
                                               ^~~~~~~~~~
src/lib.rs:41:34: 41:44 note: found defined static methods, maybe a `self` is missing?
src/lib.rs:41:34: 41:44 note: candidate #1 is defined in an impl for the type `Box<_>`
src/lib.rs:41     Box::new(telemetry_t::new()).into_raw()
                                               ^~~~~~~~~~
```
The hint in particular was confusing, since it is targeted at developers who created the impl and forgot to provide a `self` argument, rather than consumers of an impl that is specifically designed to be static.

**Comment by Manishearth** [comment]
We should probably structure this so that both suggestions are shown. Alternatively, suggest changing the method only when the method is defined in the local crate.

**Comment by birkenfeld** [comment]
Current compiler message is slightly different:
```
x.rs:3:15: 3:23 error: no method named `into_raw` found for type `Box<_>` in the current scope
x.rs:3     let y = x.into_raw();
                     ^~~~~~~~
x.rs:3:15: 3:23 note: found the following associated functions; to be used as methods, functions must have a `self` parameter
x.rs:3:15: 3:23 note: candidate #1 is defined in an impl for the type `Box<_>`
x.rs:3     let y = x.into_raw();
                     ^~~~~~~~
```
The snippet given for the "candidate #1" seems wrong, shouldn't it present the definition of `into_raw` (or, if that is not available since it's from `std`, nothing)?

**Comment by jdm** [comment]
Fixed by #33330.
